crates/biome_js_analyze/src/lint/nursery/use_components_only_module.rs

```diff
@@ -0,0 +1,347 @@
+            return exported_non_component_ids
+                .iter()
+                .filter_map(|id| {
+                    let range = id.identifier.clone().map(|x| x.range())?;
+                    Some(UseComponentsOnlyModuleState {
+                        error: ErrorType::ExportedNonComponentWithComponent,
+                        range,
+                    })
+                })
+                .collect::<Vec<UseComponentsOnlyModuleState>>();
+        }
+
+        local_component_ids
+            .map(|id| UseComponentsOnlyModuleState {
+                error: if exported_non_component_ids.is_empty() {
+                    ErrorType::UnexportedComponent
+                } else {
+                    ErrorType::NoExport
+                },
+                range: id,
+            })
+            .collect::<Vec<UseComponentsOnlyModuleState>>()
+    }
+
+    fn diagnostic(_ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        match state.error {
+            ErrorType::ExportedNonComponentWithComponent => Some(
+                RuleDiagnostic::new(
+                    rule_category!(),
+                    state.range,
+                    markup! {
+                        "Export Non-Component with components are not allowed."
+                    },
+                )
+                .note(markup! {
+                    "Fast refresh only works when a file only exports components. Use a new file to share constants or functions between components. If it is not a component, it may not be following the variable naming conventions."
+                }),
+            ),
+            ErrorType::UnexportedComponent => Some(
+                RuleDiagnostic::new(
+                    rule_category!(),
+                    state.range,
+                    markup! {
+                        "Unexported components are not allowed."
+                    },
+                )
+                .note(markup! {
+                    "Fast refresh only works when a file only exports components. Move your component(s) to a separate file. If it is not a component, it may not be following the variable naming conventions."
+                }),
+            ),
+            ErrorType::NoExport => Some(
+                RuleDiagnostic::new(
+                    rule_category!(),
+                    state.range,
+                    markup! {
+                        "Components should be exported."
+                    },
+                )
+                .note(markup! {
+                    "Fast refresh only works when a file has exports. Move your component(s) to a separate file. If it is not a component, it may not be following the variable naming conventions."
+                }),
+            )
+        }
+    }
+}
+
+// Function that returns a standard React component
+const REACT_HOOKS: [&str; 2] = ["memo", "forwardRef"];
+
+fn is_exported_react_component(any_exported_item: &ExportedItem) -> bool {
```

Comment:
Since `ExportedItem` is a public API, it makes more sense to expose a function called `exports_binding()` (just a suggestion, if you have a better name, that's great), and pass the binding name.